### ISSUE TITLE
Hydrated components cannot receive render props from spread object

### ISSUE THREAD

**Issue by natemoo-re** [solidjs/solid#1011]
### Describe the bug

Hello there! We had a user open an issue ([#3366](https://github.com/withastro/astro/issues/3366)) on [`withastro/astro`](https://github.com/withastro/astro) but it appears to be an upstream issue with Solid's hydration logic. I was able to reproduce the issue in a new SolidStart app.

This issue only seems to appear during hydration, not when using pure client-side rendering.

> Spreading props into a child component behaves oddly. If one of those props is inlined into the JSX as string content, then nothing will appear.

```jsx
import { createMutable } from 'solid-js/store';

function SpreadComp(props) {
  return (
    <div data-test-xyz={props.foo}>
      <div>this should say bar: {props.bar}</div>
      <div>this should say baz: {props.baz}</div>
    </div>
  );
}

function App() {
  const state = createMutable({
    spreadProps: {
      foo: 'foo',
      bar: 'bar',
    },
  });
  return <SpreadComp {...state.spreadProps} baz={'baz'} />;
}
```

### Your Example Website or App

https://stackblitz.com/github/natemoo-re/solid-spread-repro

### Steps to Reproduce the Bug or Issue

1. Go to `src/components/Counter.tsx`
2. Spread a nested object from `createMutable` onto another component, for example `{...state.spreadProps}`
3. Observe that `{...state.spreadProps}` are never passed to `SpreadComp`. Confirm this by logging `console.log(props)` from inside of `SpreadComp`.

### Expected behavior

As a user, I expect that the values inside of `state.spreadProps` are passed down to `SpreadComp` as props.

### Platform

- OS: macOS
- Browser: Chrome
- Version: `solid-js@1.4.2`

### Additional context

https://github.com/withastro/astro/issues/3366

cc [@samijaber](https://github.com/samijaber)

**Comment by mosheduminer** [comment]
I believe I may have observed such behavior myself as well, also with hydration, and spread props not passed. There was some discussion on discord at the time but I don't think we narrowed down the problem at the time.

**Comment by ryansolid** [comment]
Should be fixed in 1.4.3
